packages/protocol-kit/src/Safe.ts

```diff
@@ -1007,6 +1018,197 @@ class Safe {
     )
     return txResponse
   }
+
+  /**
+   * Returns the Safe Transaction encoded
+   *
+   * @async
+   * @param {SafeTransaction} safeTransaction - The Safe transaction to be encoded.
+   * @returns {Promise<string>} The encoded transaction
+   *
+   */
+  async getEncodedTransaction(safeTransaction: SafeTransaction): Promise<string> {
+    const safeSingletonContract = await getSafeContract({
+      ethAdapter: this.#ethAdapter,
+      safeVersion: await this.getContractVersion()
+    })
+
+    const encodedTransaction: string = safeSingletonContract.encode('execTransaction', [
+      safeTransaction.data.to,
+      safeTransaction.data.value,
+      safeTransaction.data.data,
+      safeTransaction.data.operation,
+      safeTransaction.data.safeTxGas,
+      safeTransaction.data.baseGas,
+      safeTransaction.data.gasPrice,
+      safeTransaction.data.gasToken,
+      safeTransaction.data.refundReceiver,
+      safeTransaction.encodedSignatures()
+    ]) as string
+
+    return encodedTransaction
+  }
+
+  /**
+   * Wraps a Safe transaction into a Safe deployment batch.
+   *
+   * This function creates a transaction batch of 2 transactions, which includes the
+   * deployment of the Safe and the provided Safe transaction.
+   *
+   * @async
+   * @param {SafeTransaction} safeTransaction - The Safe transaction to be wrapped into the deployment batch.
+   * @param {TransactionOptions} [transactionOptions] - Optional. Options for the transaction, such as from, gas price, gas limit, etc.
+   * @param {string} [customSaltNonce] - Optional. a Custom salt nonce to be used for the deployment of the Safe. If not provided, a default value is used.
+   * @returns {Promise<Transaction>} A promise that resolves to a Transaction object representing the prepared batch of transactions.
+   * @throws Will throw an error if the safe is already deployed.
+   *
+   */
+  async wrapSafeTransactionIntoDeploymentBatch(
+    safeTransaction: SafeTransaction,
+    transactionOptions?: TransactionOptions,
+    customSaltNonce?: string
+  ): Promise<Transaction> {
+    const isSafeDeployed = await this.isSafeDeployed()
+
+    // if the safe is already deployed throws an error
+    if (isSafeDeployed) {
+      throw new Error('Safe already deployed')
+    }
+
+    // we create the deployment transaction
+    const safeDeploymentTransaction = await this.createSafeDeploymentTransaction(customSaltNonce)
+
+    // First transaction of the batch: The Safe deployment Transaction
+    const firstTransaction = {
```

Comment:
better naming here to something more significative then first and second?

Reply:
done!